src/fem/finite_element.hh

```diff
@@ -39,9 +39,7 @@ template<unsigned int dim, unsigned int spacedim> class FEValuesBase;
 template<unsigned int dim, unsigned int spacedim> class FEValuesData;
 template<unsigned int dim> class FE_P_disc;
 template<unsigned int dim> class Quadrature;
-
-
-
+template<unsigned int dim, unsigned int spacedim> class XFEValues;
```

Comment:
This is too interdependent. Seems that Quadrature is not necessary. What is used by FEValues etc. should be public.
FE_P_disc should not be necessary as it is derived from FiniteElement (indirectly).

Reply:
Quadrature, FE_P_disc removed.
problem with FEValues, XFEValues left for future.